### PR TITLE
Remove use of deprecated notEmpty/allowEmpty validator methods.

### PR DESCRIPTION
**Pull request by ADmad**
Refs #485.

### CODE REVIEW

File: tests/comparisons/Model/testBakeTableValidation.php [hunk 1]
```diff
@@ -44,32 +44,33 @@ public function validationDefault(Validator $validator)
...
             ->requirePresence('name', 'create')
-            ->notEmpty('name')
-            ->maxLength('name', 100, 'Name must be shorter than 100 characters.');
+            ->notEmptyString('name');
 
         $validator
             ->nonNegativeInteger('count')
             ->requirePresence('count', 'create')
-            ->notEmpty('count');
+            ->notEmptyString('count');
```
Comment (dereuromark) [inline tests/comparisons/Model/testBakeTableValidation.php]: string for ints is a bit weird.
can we use ...Scalar here etc?

Comment (markstory) [reply tests/comparisons/Model/testBakeTableValidation.php]: From what I remember the thinking was that POST data from forms is always strings.

File: src/Shell/Task/ModelTask.php [hunk 2]
```diff
@@ -781,6 +793,34 @@ public function fieldValidation($schema, $fieldName, array $metaData, $primaryKe
...
+            case 'datetime':
+            case 'timestamp':
+                return 'DateTime';
+        }
+
+        if (preg_match('/file|image/', $fieldName)) {
+            return 'File';
+        }
+
+        return 'String';
```
Comment (dereuromark) [inline src/Shell/Task/ModelTask.php]: The alternative would be to use normal "camelBacked" (string, dateTime) and ufcirst() the result as suffix.
This way the "type" return here is true. But no big deal either way.

File: tests/comparisons/Model/testBakeTableValidation.php (hunk 1 above)
Comment (dereuromark) [reply tests/comparisons/Model/testBakeTableValidation.php]: But we know that this is not true for this validation process. Marshalled data in general can also come from webservice, and other endpoints, as such they sure can be in a nonstring format already.

Comment (ADmad) [reply tests/comparisons/Model/testBakeTableValidation.php]: Yes data from webservices can be in non string format but if a value of valid type is set for a non string field then it is non empty anyway and "not empty" check will pass as expected.

Taking the example of an integer field only way to indicate empty value is passing `null` or empty string for it, which is covered by using `notEmptyString()`. (All `notEmpty*` methods treat `null` as empty as expected).

Comment (dereuromark) [reply tests/comparisons/Model/testBakeTableValidation.php]: I didnt mean null, but but the actual value (int, float, ..).
So for me scalar still sounds more correct than string.
We might have to patch the core here.

Comment (ADmad) [reply tests/comparisons/Model/testBakeTableValidation.php]: Bake can be updated if/when core is updated. For now this is good to go.

File: src/Shell/Task/ModelTask.php [hunk 3]
```diff
@@ -766,6 +757,27 @@ public function fieldValidation($schema, $fieldName, array $metaData, $primaryKe
...
+                'rule' => 'allowEmpty' . $this->getEmptyMethodType($fieldName, $metaData),
+                'args' => [],
+            ];
+        } else {
+            $validation['requirePresence'] = [
+                'rule' => 'requirePresence',
+                'args' => ["'create'"],
+            ];
+            $validation['notEmpty'] = [
+                'rule' => 'notEmpty' . $this->getEmptyMethodType($fieldName, $metaData),
```
[the remark this answers is not in the document]
Comment (ADmad) [reply src/Shell/Task/ModelTask.php]: Woops, don't know how I missed that.

**Comment by ADmad** [comment]
@markstory I have fixed the method calls.

**Pull request merged**